### PR TITLE
migrate ros2 devel

### PR DESCRIPTION
**Pull request by Karsten1987**
This is a first PR to import the work done by @bponsler and @MarcTestier. 

I am trying to be non-intrusive in terms of style, but I am happy to run the batch of default ros2 linters. I leave that decision though to @tonybaltovski.

### CODE REVIEW

**Comment by tonybaltovski** [comment]
@Karsten1987 the default ros2 linters sound good to me.

**Comment by Karsten1987** [comment]
I tried to enable the default linters (cpplint, uncrustify, etc.) but given the amount of utf-8 character and such applying the linters is basically impossible.
That being said, I'd leave the PR as-is and enable the linters on the follow up packages.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,34 +1,27 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(urg_c)
 
+# Support C++14
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" -std=c++14)
```
Comment (clalancette) [inline CMakeLists.txt]: I'll suggest that you use the more standard template that we've been using in ROS 2 for setting this, like: https://github.com/ros2/demos/blob/857c436e76bc6b40d32f096f6989bf51bb73f359/image_tools/CMakeLists.txt#L5

File: current/src/urg_time.c [hunk 2]
```diff
@@ -20,8 +20,8 @@ void urg_walltime(unsigned long *sec, unsigned long *nsec)
 #else
     struct timeval timeofday;
     gettimeofday(&timeofday,NULL);
-    sec  = timeofday.tv_sec;
-    nsec = timeofday.tv_usec * 1000;
+    *sec  = (unsigned long)timeofday.tv_sec;
```
Comment (clalancette) [inline current/src/urg_time.c]: Suggest using a C++-style cast here.

File: CMakeLists.txt (hunk 1 above)
Comment (Karsten1987) [reply CMakeLists.txt]: done in https://github.com/ros-drivers/urg_c/pull/11/commits/8fd5611deb0cfca0a848d63c9cad70a7844401e7

File: current/src/urg_time.c (hunk 2 above)
Comment (Karsten1987) [reply current/src/urg_time.c]: done in https://github.com/ros-drivers/urg_c/pull/11/commits/83e68fb8d370a234b9b49018cdbb18c33f6f396f

**Comment by Karsten1987** [comment]
had to remove the static cast as it's a C-only library.

**Comment by clalancette** [comment]
> had to remove the static cast as it's a C-only library.

Oh, haha, my bad.  I didn't realize that.  Looks good :+1:

**Comment by clalancette** [comment]
@chadrockey @tonybaltovski This looks good to me now, and is passing CI.  I think this is ready to merge.  Please give it a look and let us know of any comments you have.  If I don't hear anything by the end of the week, I'll go ahead and merge and we can follow-up later.

**Comment by clalancette** [comment]
I'm going to go ahead and merge this, but as always, feel free to leave comments and we can follow-up later.

**Pull request merged**